docker/compose/notebooks/utils.ipynb

```diff
@@ -0,0 +1,223 @@
+{
+ "cells": [
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "id": "3c22eccc-8655-4b87-bc0d-cd5a068b889b",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#  Licensed to the Apache Software Foundation (ASF) under one\n",
+    "#  or more contributor license agreements.  See the NOTICE file\n",
+    "#  distributed with this work for additional information\n",
+    "#  regarding copyright ownership.  The ASF licenses this file\n",
+    "#  to you under the Apache License, Version 2.0 (the\n",
+    "#  \"License\"); you may not use this file except in compliance\n",
+    "#  with the License.  You may obtain a copy of the License at\n",
+    "#\n",
+    "#      http://www.apache.org/licenses/LICENSE-2.0\n",
+    "#\n",
+    "#  Unless required by applicable law or agreed to in writing, software\n",
+    "#  distributed under the License is distributed on an \"AS IS\" BASIS,\n",
+    "#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.\n",
+    "#  See the License for the specific language governing permissions and\n",
+    "# limitations under the License."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "id": "f8bbcda9-7b8a-4415-8915-acadcfa07660",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import os\n",
+    "from pyspark.sql import SparkSession"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "id": "c44b9eca-63e2-468a-ba86-8113533b4d15",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "HUDI_JAR = os.environ.get(\"HUDI_SPARK_BUNDLE\")\n",
+    "if not HUDI_JAR:\n",
+    "        raise EnvironmentError(\"HUDI_SPARK_BUNDLE environment variable not set\")\n",
+    "\n",
+    "HADOOP_S3_JAR = \"/opt/spark/jars/hadoop-aws-3.3.4.jar,/opt/spark/jars/aws-java-sdk-bundle-1.12.734.jar\"\n",
+    "\n",
+    "ALL_JARS = f\"{HUDI_JAR},{HADOOP_S3_JAR}\""
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "id": "7f104543-3374-4c78-b666-bf44f3be962c",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def get_spark_session(app_name=\"Hudi-Notebooks\"):\n",
```

Comment:
utils need to be put in a notebook? they're supposed to be just imported by other notebooks and can live in .py files?

Reply:
we should avoid putting redundant code in the codebase